lib/model/app_model/list_invoices.dart

```diff
@@ -9,80 +10,88 @@ class AppListInvoices {
-        return AppListInvoices();
+        list.add(temp);
       }
-      return AppListInvoices(invoice: appInvoices as List<AppInvoice>);
+      return AppListInvoices(invoice: list);
     } else {
       return AppListInvoices();
     }
   }
 }
 
 class AppInvoice {
-  /// payment hash
-  final String paymentHash;
-
   /// bolt11 identifier
   final String bolt11;
 
   /// The quantity of Bitcoin in millisatoshi
   final String amount;
 
-  /// If the invoice is paid or not on the blockchain
-  final String status;
+  final String paymentRequest;
 
-  /// The Invoice paid time
-  final String paidTime;
+  final String prefix;
 
-  /// The Invoice description
-  final String label;
+  final List<int> signature;
 
-  /// The Invoice description
-  final String description;
+  final List<pay_req.TaggedField> tags;
+
+  final String timeStamp;
 
-  /// flag to identify invoice
   final String identifier;
 
+  final String status;
+
   AppInvoice(
-      {required this.paymentHash,
+      {required this.bolt11,
       required this.amount,
+      required this.paymentRequest,
+      required this.prefix,
+      required this.signature,
+      required this.tags,
+      required this.timeStamp,
       required this.status,
-      required this.paidTime,
-      required this.bolt11,
-      required this.description,
-      required this.label,
       this.identifier = "invoice"});
 
   factory AppInvoice.fromJSON(Map<String, dynamic> json,
       {bool snackCase = false, bool isObject = false}) {
     LogManager.getInstance.debug("$json");
-    var bolt11 = json.withKey("bolt11", snackCase: snackCase) ??
-        json.withKey("bolt12", snackCase: snackCase);
-    var paymentHash = json.withKey("paymentHash", snackCase: snackCase);
+    var bolt11 = json.withKey("bolt11", snackCase: snackCase);
     var status = json.withKey("status", snackCase: snackCase);
-    var received = json.parseMsat(
-        key: "amountReceivedMsat", snackCase: snackCase, isObject: isObject);
-    received.toString();
-    var paidAt = json.withKey("paidAt", snackCase: snackCase);
-    var description = json.withKey("description", snackCase: snackCase);
-    var label = json.withKey("label", snackCase: snackCase);
+    pay_req.Bolt11PaymentRequest req = pay_req.Bolt11PaymentRequest(bolt11);
+    LogManager.getInstance.debug(
+        "NEW RESPONSE HERE : ${req.amount} ${req.paymentRequest} ${req.prefix} ${req.signature} ${req.tags} ${req.timestamp}");
+    var amount = req.amount;
+    var paymentRequest = req.paymentRequest;
+    var pre = req.prefix;
+    String prefix = '';
+
+    /// Tells us about the prefix of the payment request whether it is of lightning invoice or other.
+    if (pre == pay_req.PayRequestPrefix.lntb) {
+      prefix = "lntb";
+    } else if (pre == pay_req.PayRequestPrefix.lnbcrt) {
+      prefix = "lncbrt";
+    } else if (pre == pay_req.PayRequestPrefix.lnbc) {
+      prefix = "lnbc";
+    } else {
+      prefix = "lnsb";
+    }
```

Comment:
Ah ok I got it, we do not cate about the prefix inside the UI! it is uggly no?

Reply:
Yes, it is kind of ugly. I'll change it.